pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -101,67 +102,59 @@ func (r *ReconcileCheInstallation) Reconcile(request reconcile.Request) (reconci
 		return reconcile.Result{}, err
 	}
 
-	err = r.ensureCheInstallation(reqLogger, cheInstallation)
-	return reconcile.Result{}, err
-}
-
-func (r *ReconcileCheInstallation) ensureCheInstallation(logger logr.Logger, cheInstallation *v1alpha1.CheInstallation) error {
-	ns := cheInstallation.Spec.CheOperatorSpec.Namespace
-
-	if created, err := r.ensureCheNamespace(logger, cheInstallation); err != nil {
-		return r.wrapErrorWithStatusUpdate(logger, cheInstallation, r.setStatusCheSubscriptionFailed, err, "failed to create namespace %s", ns)
+	if ns, created, err := r.ensureCheNamespace(reqLogger, cheInstallation); err != nil {
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, cheInstallation, r.setStatusCheSubscriptionFailed, err, "failed to create namespace %s", cheInstallation.Spec.CheOperatorSpec.Namespace)
 	} else if created {
-		return nil
+		return reconcile.Result{}, nil
+	} else if ns.Status.Phase != v1.NamespaceActive {
```

Comment:
This check is more appropriate to do in `ensureCheNamespace()` .. bcoz if in future we add any code at end of this method [here](https://github.com/codeready-toolchain/toolchain-operator/blob/8ac8a996d5dc131b319cd1471d9e52a37d1ea020/pkg/controller/cheinstallation/cheinstallation_controller.go#L150) we really need to make that Namespace should be `Active` and then we adding more logic in `ensureCheNamespace()`.

Reply:
ok, let me see what I can do

Reply:
code refactored in 0ab8b1f